src/Playwright/Core/BrowserContext.cs

```diff
@@ -245,7 +245,7 @@ public async Task<IPage> NewPageAsync()
         }
 
         public Task RouteAsync(string url, Action<IRoute> handler, BrowserContextRouteOptions options = default)
-            => RouteAsync(url, null, null, handler, options);
+            => RouteAsync(url, new Regex(CombineUrlWithBase(url).GlobToRegex()), null, handler, options);
```

Comment:
Before the change, `RouteAsync` supported either glob (first argument), regex (second argument) or a callback (third argument). You are folding the glob one into regex, which is great. With that, the first argument (glob) should go away (line :364 new code). Along with the `RouteSetting`'s `Url` field.

Reply:
Glob/Url ist still needed so the user can `Unroute` it with the same glob/url.

Reply:
Not a C# expert, but I was hoping there was a way to compare those for equality. If not, we could compare underlying patterns + flags?

Reply:
Ah you mean that in `UnrouteAsync` a regex is also created from the glob and checked for equality with the route regex?
Then no the regex objects itself won't be equal but their `ToString` representations are:

```c#
var url = @"^http:\/\/localhost:8909\/empty\.html$";
var regex1 = new Regex(url);
var regex2 = new Regex(url);

Console.WriteLine(regex1 == regex2); // false
Console.WriteLine(regex1.Equals(regex2)); // false
Console.WriteLine(regex1.ToString() == regex2.ToString() && regex1.Options == regex2.Options); // true
```
With that we could get rid of the `Url` property, which I have done in https://github.com/microsoft/playwright-dotnet/pull/1935/commits/300bcc2479ea1e6e40505d25da41ea0d132891d9